src/CommandLineUtils/Internal/ValueParsers/ValueParser.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Nate McMaster.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace McMaster.Extensions.CommandLineUtils.Abstractions
+{
+    using System;
+    using System.Globalization;
+
+    /// <summary>
+    /// Provides methods for creating <see cref="IValueParser{T}"/>
+    /// boilerplate implementations.
+    /// </summary>
+
+    public static class ValueParser
```

Comment:
Can we name this `ValueParserFactory`?

Reply:
I was following in the footsteps of giants, to call out a few examples:

- [`StringComparer.Create`](https://docs.microsoft.com/en-us/dotnet/api/system.stringcomparer.create?view=netcore-2.1)
- [`Tuple.Create`](https://docs.microsoft.com/en-us/dotnet/api/system.tuple.create?view=netcore-2.1)
- [`ValueTuple.Create`](https://docs.microsoft.com/en-us/dotnet/api/system.valuetuple.create?view=netcore-2.1)
- [`Partitioner.Create`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.partitioner.create?view=netcore-2.1)

There's a lot lesser examples of types with `Factory` in their name. I can only think of [`TaskFactory`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskfactory?view=netcore-2.1).

`ValueParser` keeps it plain and simple but if you still insist, I have no qualms about renaming it to `ValueParserFactory`.

Reply:
Ok, `ValueParser` it is :)